r2-netty/src/main/java/com/linkedin/r2/netty/client/http2/Http2ChannelPoolFactory.java

```diff
@@ -116,8 +119,19 @@ public Http2ChannelPoolFactory(
     _maxContentLength = maxContentLength;
     _tcpNoDelay = tcpNoDelay;
 
-    Bootstrap bootstrap = !org.apache.commons.lang.StringUtils.isEmpty(udsAddress) ?
-        new Bootstrap().channel(EpollDomainSocketChannel.class): new Bootstrap().channel(NioSocketChannel.class);
+    final Class channelClass;
+    if (org.apache.commons.lang.StringUtils.isEmpty(udsAddress)) {
+      channelClass = NioSocketChannel.class;
+    } else {
+      if (Epoll.isAvailable()) {
+        channelClass = EpollDomainSocketChannel.class;
+      } else if (KQueue.isAvailable()){
+        channelClass = KQueueDomainSocketChannel.class;
+      } else {
+        throw new IllegalStateException("Neither Epoll or Kqueue domain socket transport available");
+      }
+    }
+    Bootstrap bootstrap = new Bootstrap().channel(channelClass);
```

Comment:
Can we extract the above logic into a util and deduplicate the code?

Reply:
Yeah that was a bit of a smell. Moved to a common location. Thanks!